ddtrace/contrib/algoliasearch/patch.py

```diff
@@ -18,23 +18,41 @@
     collect_query_text=False
 ))
 
+algoliasearch_version = tuple([int(i) for i in VERSION.split('.')])
+
 
 def patch():
     if getattr(algoliasearch, DD_PATCH_ATTR, False):
         return
 
     setattr(algoliasearch, '_datadog_patch', True)
-    _w(algoliasearch.index, 'Index.search', _patched_search)
-    Pin(
+
+    pin = Pin(
         service=config.algoliasearch.service_name, app=APP_NAME,
         app_type=AppTypes.db
-    ).onto(algoliasearch.index.Index)
+    )
+    if algoliasearch_version < (2, 0) and algoliasearch_version >= (1, 0):
```

Comment:
I'd rather go with `try/import/except ImportError` rather than version checking. It's more likely to work without a change for future version (e.g. `3.0`) if that ever happens.

Reply:
I would advocate for a `try/import/except` too.

Reply:
I'll go with a hybrid approach, I'll surround this statement:

```
try:
    from algoliasearch.version import VERSION
except ImportError:
    <... abort patching ...>
```

And continue using the version string to determine code paths. The rationale is that the version provides us with a consistent way of determining exactly which functions we want to patch without having to worry about possible commonalities in structure of modules. E.g. if version 2 and version 3 both have the following import succeeding:

```
from algoliasearch.search_index import SearchIndex
```

Yet the signature of the function that does the searching `SearchIndex.search` is different (i.e. `def search(self, query, query_args, request_options)` vs `def search(self, query, request_options)` -- this change actually happened between 1.2 and 2) the try/except approach won't provide sufficient protections.